src/pages/spawn/spawnVolume/spawnVolumeTable/SpawnVolumeTableActions.tsx

```diff
@@ -1,38 +1,53 @@
+import { useQuery } from "@apollo/client";
 import styled from "@emotion/styled";
 import { size } from "constants/tokens";
+import { MyHostsQuery, MyHostsQueryVariables } from "gql/generated/types";
+import { GET_MY_HOSTS } from "gql/queries";
 import { MyVolume } from "types/spawn";
 import { DeleteVolumeBtn } from "./spawnVolumeTableActions/DeleteVolumeBtn";
 import { EditButton } from "./spawnVolumeTableActions/EditButton";
+import { MigrateBtn } from "./spawnVolumeTableActions/MigrateBtn";
 import { MountBtn } from "./spawnVolumeTableActions/MountBtn";
 import { UnmountBtn } from "./spawnVolumeTableActions/UnmountBtn";
 
 interface Props {
   volume: MyVolume;
 }
 
-export const SpawnVolumeTableActions: React.VFC<Props> = ({ volume }) => (
-  <FlexRow>
-    <DeleteVolumeBtn
-      data-cy={`trash-${volume.displayName || volume.id}`}
-      volume={volume}
-    />
-    {volume.host ? (
-      <UnmountBtn
-        data-cy={`unmount-${volume.displayName || volume.id}`}
+export const SpawnVolumeTableActions: React.VFC<Props> = ({ volume }) => {
+  const { data: myHostsData } = useQuery<MyHostsQuery, MyHostsQueryVariables>(
+    GET_MY_HOSTS
+  );
```

Comment:
Although apollo is smart enough to not duplicate this query can we still move it up and pass it in as a prop.